### PR TITLE
PERMANOVA: by = margin, tasks, CLR

### PR DESCRIPTION
**Pull request by TuomasBorman**
https://github.com/microbiome/OMA/issues/171

### CODE REVIEW

**Review by antagomir — COMMENTED** [review]
Shall we also add 1-2 sentence explanation why adonis2+"Group" is recommended?

File: 20_beta_diversity.Rmd [hunk 1]
```diff
@@ -498,16 +498,60 @@ different community composition.
...
+set.seed(75)
+adonis2(t(assay(enterotype ,"relabundance"))  ~ Nationality + Gender + ClinicalStatus,
+        data = colData(enterotype ),
+        permutations = 99)
+
+# Multiple variables, default: by = "terms"
+set.seed(75)
+adonis2(t(assay(enterotype ,"relabundance"))  ~ ClinicalStatus + Nationality + Gender,
+        data = colData(enterotype),
+        permutations = 99)
```
Comment (antagomir) [inline 20_beta_diversity.Rmd:546]: This is informative but I am wondering if it is a bit much, to keep OMA compact? One option would be to add here "Task" suggesting that the user implements & tries this out themselves (just mention function names and changing variable order compared to the first example). Or this comparison could be on additional supplementary page and here just a link.

File: 20_beta_diversity.Rmd [hunk 2]
```diff
@@ -498,16 +498,60 @@ different community composition.
...
+```
+
+We can perform PERMANVA with `adonis2` function or by first performing distance-based
```
Comment (antagomir) [inline 20_beta_diversity.Rmd:549]: PERMNVA->PERMANOVA

File: 20_beta_diversity.Rmd [hunk 3]
```diff
@@ -498,16 +498,60 @@ different community composition.
...
+# Multiple variables, default: by = "terms"
+set.seed(75)
+adonis2(t(assay(enterotype ,"relabundance"))  ~ ClinicalStatus + Nationality + Gender,
+        data = colData(enterotype),
+        permutations = 99)
+```
+
+We can perform PERMANVA with `adonis2` function or by first performing distance-based 
+redundancy analysis (dbRDA), and then applying permutational test for result of 
+redundancy analysis. Advantage ot hthe latter approach is that by doing so we can get
```
Comment (antagomir) [inline 20_beta_diversity.Rmd:551]: hthe

**Comment by TuomasBorman** [comment]
True, I addedthat to extras for now

**Review by antagomir — APPROVED** [review]

File: 98_exercises.Rmd [hunk 4]
```diff
@@ -54,7 +64,7 @@ Functions that might be useful: wilcox.test, kruskal.test, ggplot, pheatmap, Com
 
 Functions that might be useful: runMDS, runRDA, anova.cca, transformSamples, agglomerateByRank, ggplot, plotReducedDim
 
-## Introduction to MAE
+## Introduction to MAE {.unlisted .unnumbered}
```
Comment (antagomir) [inline 98_exercises.Rmd:67]: Shall we use full names + acronym like MultiAssayExperiment (MAE). Could be more clear?

**Review by antagomir — APPROVED** [review]
Looks good!

File: 11_taxonomic_information.Rmd [hunk 5]
```diff
@@ -182,7 +182,8 @@ For complete list of available transformations and parameters, see function
 [help](https://microbiome.github.io/mia/reference/transformCounts.html).
 
 ```{r}
-tse <- transformSamples(x = tse, abund_values = "counts", method = "clr", 
+tse <- transformSamples(tse, method = "relabundance")
+tse <- transformSamples(x = tse, abund_values = "relabundance", method = "clr", 
                         pseudocount = 1, name = "clr_transformation")
```
Comment (TuomasBorman) [inline 11_taxonomic_information.Rmd:187]: 1 They should give equal result, but we should probably pick which one we use: pseudocount in relabundance step or in clr step

File: 19_visualization_techniques.Rmd [hunk 6]
```diff
@@ -287,7 +287,8 @@ tse_phylum <- agglomerateByRank(tse,
                                 rank = "Phylum",
                                 onRankOnly = TRUE)
 # Add clr-transformation on samples
-tse_phylum <- transformSamples(tse_phylum, method = "clr", pseudocount = 1)
+tse_phylum <- transformSamples(tse_phylum, method = "relabundance", pseudocount = 1)
+tse_phylum <- transformSamples(tse_phylum, method = "clr", abund_values = "relabundance")
```
Comment (TuomasBorman) [inline 19_visualization_techniques.Rmd:291]: 2

**Review by antagomir — APPROVED** [review]

**Pull request merged**